src/app/(marketing)/docs/[slug]/FeaturePageClient.tsx

```diff
@@ -0,0 +1,181 @@
+                                                                            height={1000}
+                                                                            className="w-full h-full border rounded-md border-neutral-200 shadow-md"
+                                                                        />
+                                                                        <p className="text-sm text-neutral-500 mt-2">
+                                                                            {image.caption}
+                                                                        </p>
+                                                                    </div>
+                                                                ))}
+                                                            </div>
+                                                        )}
+                                                    </div>
+                                                </div>
+                                            );
+                                        },
+                                    )}
+                                </div>
+                            ))}
+                        </div>
+                    )}
+                </div>
+            </div>
+
+            {/* Mini Navigation - Only show if more than one section */}
+            {feature.howToUse && feature.howToUse.length > 1 && (
+                <div className="hidden lg:block w-64 flex-shrink-0">
+                    <div className="sticky top-24">
+                        <div className="p-4">
+                            <h3 className="text-sm font-semibold text-neutral-900 mb-3">
+                                Contents
+                            </h3>
+                            <nav className="space-y-2">
+                                {feature.howToUse.map((section: any, index: number) => (
+                                    <button
+                                        key={index}
+                                        onClick={() => scrollToSection(index)}
+                                        className={cn(
+                                            "block w-full text-left text-sm rounded-md transition-colors",
+                                            activeSection === index
+                                                ? "text-neutral-800 underline"
+                                                : "text-neutral-500 hover:text-neutral-900"
+                                        )}
+                                    >
+                                        {section.title || `Section ${index + 1}`}
+                                    </button>
+                                ))}
+                            </nav>
+                        </div>
+                    </div>
+                </div>
+            )}
+        </div>
+    );
+}
```

Comment:
This is (for me) a component that could be divided into smaller parts for readability.

For starters, table of contents aka mini navigation could be a separate component, and a single Step could be a separate component.

I think it would make this file much easier to scan :)